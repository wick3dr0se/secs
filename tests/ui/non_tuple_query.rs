use secs::prelude::*;

fn optional_components() {
    let mut world = World::default();

    world.query::<u32>(|_, _| {});
    //~^ ERROR: `u32` is not a valid query
}
