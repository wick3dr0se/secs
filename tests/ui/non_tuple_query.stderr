error[E0277]: `u32` is not a valid query
 --> tests/ui/non_tuple_query.rs:6:11
  |
6 |     world.query::<u32>(|_, _| {});
  |           ^^^^^
  |
  = help: the trait `secs::query::Query<'_>` is not implemented for `u32`
  = note: only tuples with 1 or up to 5 elements can be used as queries
  = help: the following other types implement trait `secs::query::Query<'a>`:
            (T, U)
            (T, U, V)
            (T, U, V, W)
            (T, U, V, W, X)
            (T,)

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0277`.
